bug(python): failed to infer column name from the schema

### LanceDB version

v0.13.0

### What happened?

Query except FTS requires vector column.
Failed to infer vector column at 
https://github.com/lancedb/lancedb/blob/main/python/python/lancedb/table.py#L1635
the vector_column_name is None without throwing exception.

Proposal
- early error out if failed to infer vector column name from schema
- check why schema does not have vector column

### Are there known steps to reproduce?

Test dataset
```
def _test_search_table() -> pa.Table:
    return pa.Table.from_pandas(
        pd.DataFrame(
            {
                "item_id": [1, 2, 3, 4, 5],
                "inner_id": [1, 2, 3, 4, 5],
                "category": ["a", "a", "b", "b", "c"],
                "numeric_int": [2, 5, 3, 1, 4],
                "numeric_float": [0.2, 0.5, 0.3, 0.1, 0.4],
                "category_set": [
                    ["d", "e"],
                    ["d", "f"],
                    ["d", "g"],
                    ["g", "h"],
                    ["h", "i"],
                ],
            }
        )
    )
```
with  search query
```
result = (
    self._table_inner.search(query_type="hybrid")
    .text(query_text)
    .vector(text_query_encoder([query_text])[0])
    .rerank(reranker=LinearCombinationReranker(text_semantic_search_weight))
    .select([self.INNER_ID_NAME])
    .limit(k)
    .to_arrow()
    .to_pylist()
)
```
throws exception
```
in text_semantic_search
    .to_arrow()
../../.conda/envs/dev/lib/python3.11/site-packages/lancedb/query.py:1046: in to_arrow
    vector_results = vector_future.result()
../../.conda/envs/dev/lib/python3.11/concurrent/futures/_base.py:449: in result
    return self.__get_result()
../../.conda/envs/dev/lib/python3.11/concurrent/futures/_base.py:401: in __get_result
    raise self._exception
../../.conda/envs/dev/lib/python3.11/concurrent/futures/thread.py:58: in run
    result = self.fn(*self.args, **self.kwargs)
../../.conda/envs/dev/lib/python3.11/site-packages/lancedb/query.py:647: in to_arrow
    return self.to_batches().read_all()
../../.conda/envs/dev/lib/python3.11/site-packages/lancedb/query.py:678: in to_batches
    result_set = self._table._execute_query(query, batch_size)
../../.conda/envs/dev/lib/python3.11/site-packages/lancedb/table.py:1742: in _execute_query
    return ds.scanner(
../../.conda/envs/dev/lib/python3.11/site-packages/lance/dataset.py:369: in scanner
    builder = builder.nearest(**nearest)
../../.conda/envs/dev/lib/python3.11/site-packages/lance/dataset.py:2441: in nearest
    if self.ds.schema.get_field_index(column) < 0:
pyarrow/types.pxi:3012: in pyarrow.lib.Schema.get_field_index
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: expected bytes, NoneType found
```
schema.get_field_index(column)  throws exception

pyarrow schema
```
'inner_id: int64, has_vector: bool, vector: fixed_size_list<item: double>[384], item_id: int64, category: string, numeric_int: int64, numeric_float: double, category_set: list<item: string>'
```

column is None
```
(Pdb) pp nearest
{'column': None,
 'k': 3,
 'metric': 'L2',
 'nprobes': 20,
 'q': <vector_values>,
 'refine_factor': None
}
```

After pass in search(query, vector_column_name), the test passed.

--- a/python/python/lancedb/util.py
+++ b/python/python/lancedb/util.py
@@ -308,12 +308,16 @@ def infer_vector_column_name(
         return None
 
     if query is not None or query_type == "hybrid":
-        try:
-            vector_column_name = inf_vector_column_query(
-                schema, dim=_query_vector_dim(query)
-            )
-        except Exception as e:
-            raise e
+        vector_column_name = inf_vector_column_query(
+            schema, dim=_query_vector_dim(query)
+        )
+
+    if vector_column_name is None:
+        raise ValueError(
+            "No vector column found in the schema. Please specify the "
+            "vector column name explicitly via the `vector_column_name` "
+            "parameter."
+        )
 
     return vector_column_name
 

--- a/python/python/lancedb/util.py
+++ b/python/python/lancedb/util.py
@@ -307,10 +307,12 @@ def infer_vector_column_name(
         # FTS queries do not require a vector column
         return None
 
-    if query is not None or query_type == "hybrid":
-        vector_column_name = inf_vector_column_query(
-            schema, dim=_query_vector_dim(query)
-        )
+    if query is None and query_type != "hybrid":
+        # No vector search was requested (e.g. a plain scan), so there's
+        # nothing to infer.
+        return None
+
+    vector_column_name = inf_vector_column_query(schema, dim=_query_vector_dim(query))
 
     if vector_column_name is None:
         raise ValueError(

--- a/python/python/tests/test_util.py
+++ b/python/python/tests/test_util.py
@@ -924,3 +924,23 @@ def test_sanitize_data_stream():
 
     with pytest.raises(ValueError):
         next(output)
+
+
+def test_infer_vector_column_raises_clear_error(tmp_path):
+    """Regression: querying a table with no inferable vector column should raise
+    a clear ValueError, not a cryptic TypeError (issue #1653).
+
+    Previously, inf_vector_column_query silently returned None which then caused
+    a confusing TypeError deep in schema lookup. The fix adds a ValueError guard
+    so the user gets a direct, actionable error message.
+    """
+    db = lancedb.connect(tmp_path)
+    table = db.create_table(
+        "no_vec",
+        data=[{"id": 1, "text": "hello"}, {"id": 2, "text": "world"}],
+    )
+
+    with pytest.raises(ValueError, match="vector"):
+        # Plain vector search on a table with no vector column should raise
+        # a clear ValueError, not a cryptic TypeError.
+        table.search([1.0, 2.0]).to_list()

